### PR TITLE
China adaptation

### PR DESCRIPTION
**Pull request by azeliashchonak**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -130,10 +140,12 @@ resource "aws_route53_record" "alb" {
     zone_id                = "${module.alb.load_balancer_zone_id}"
     evaluate_target_health = true
   }
+
+  count = "${data.aws_partition.current.partition == "aws" ? 1 : 0}"
 }
 
 resource "aws_route53_record" "alb-subdomain" {
-  count   = "${ var.enable_subdomains ? 1 : 0 }"
+  count   = "${var.enable_subdomains == true ? "${data.aws_partition.current.partition == "aws" ? 1 : 0}" : 0 }"
```
Comment (ToROxI) [inline main.tf]: As far as we don't support subdomains in china (no matter on module input parameters):
```suggestion
  count   = "${data.aws_partition.current.partition == "aws" ? "${var.enable_subdomains == true ? 1 : 0}" : 0 }"
```

File: outputs.tf [hunk 2]
```diff
@@ -25,7 +25,7 @@ output "alb_dns_name" {
 
 output "alb_custom_dns_name" {
   description = "The custom DNS name of the load balancer."
-  value       = "${aws_route53_record.alb.name}"
+  value       = "${data.aws_partition.current.partition == "aws" ? "${element(concat(aws_route53_record.alb.*.name, list("")), 0)}":"${element(concat(data.aws_iam_server_certificate.ss_cert.*.name, list("")), 0)}"}"
```
Comment (ToROxI) [inline outputs.tf]:
```suggestion
  value       = "${element(concat(aws_route53_record.alb.*.name, data.aws_iam_server_certificate.ss_cert.*.name, list("")), 0)}"
```

File: outputs.tf [hunk 3]
```diff
@@ -79,5 +79,5 @@ output "alb_target_group_names" {
 }
 
 output "root_domain_hosted_zone_id" {
-  value = "${data.aws_route53_zone.alb.zone_id}"
+  value = "${data.aws_partition.current.partition == "aws" ? "${element(concat(aws_route53_record.alb.*.zone_id, list("")), 0)}":""}"
```
Comment (ToROxI) [inline outputs.tf]:
```suggestion
  value = "${element(concat(aws_route53_record.alb.*.zone_id, list("")), 0)}"
```

**Review by ToROxI — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (azeliashchonak) [reply main.tf]: fixed

File: outputs.tf (hunk 3 above)
Comment (azeliashchonak) [reply outputs.tf]: fixed

File: outputs.tf (hunk 2 above)
Comment (azeliashchonak) [reply outputs.tf]: fixed

**Review by ToROxI — APPROVED** [review]

**Pull request merged**